octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -73,12 +88,146 @@ async def initialize_impl(self):
+        """
+        parser = self.orders_parser(self.exchange_manager.exchange)
+        return await parser.parse_orders(raw_orders, order_type=order_type, quantity=quantity,
+                                         price=price, status=status, symbol=symbol, side=side,
+                                         timestamp=timestamp, check_completeness=check_completeness)
+
+    async def parse_trade(self, raw_trade: dict, check_completeness: bool = True) -> dict:
+        """
+        use this method to parse a single trade
+
+        :param raw_trade:
+
+        optional:
+        :param status: to use if it's missing in the trade
+        :param trade_type: to use if it's missing in the trade
+        :param price: to use if it's missing in the trade
+        :param quantity: to use if it's missing in the trade
+        :param symbol: to use if it's missing in the trade
+        :param side: to use if it's missing in the trade
+        :param timestamp: to use if it's missing in the trade
+
+        :param check_completeness: if true checks all attributes, 
+            if somethings missing it'll try to fetch it from the exchange
+
+        :return: formatted trade dict (100% complete or we raise NotImplemented)
+        """
+        parser = self.trades_parser(self.exchange_manager.exchange)
+        return await parser.parse_trade(raw_trade, check_completeness=check_completeness)
+
+    async def parse_trades(self, raw_trades, check_completeness: bool = True) -> list:
+        """
+        use this method to format a list of trade dicts
+
+        :param raw_trade: raw trades with eventually missing data
+
+        optional:
+        :param status: to use if it's missing in the trade
+        :param trade_type: to use if it's missing in the trade
+        :param price: to use if it's missing in the trade
+        :param quantity: to use if it's missing in the trade
+        :param symbol: to use if it's missing in the trade
+        :param side: to use if it's missing in the trade
+        :param timestamp: to use if it's missing in the trade
+
+        :param check_completeness: if true checks all attributes, 
+            if somethings missing it'll try to fetch it from the exchange
+
+        :return: formatted trades list (100% complete or we raise NotImplemented report)
+            
+        """
+        parser = self.trades_parser(self.exchange_manager.exchange)
+        return await parser.parse_trades(raw_trades, check_completeness=check_completeness)
+
+    async def parse_position(self, raw_position: dict) -> dict:
+        """
+        use this method to parse a single position
+
+        :param raw_position:
+
+        :param check_completeness: if true checks all attributes, 
+            if somethings missing it'll try to fetch it from the exchange
+
+        :return: formatted position dict (100% complete or we raise NotImplemented)
+        """
+        parser = self.positions_parser(self.exchange_manager.exchange)
```

Comment:
Shouldn't we have create a parser instance once and for all instead of creating a new one for each call?

Reply:
Yeah I had that, but we use self a lot to store temp data. So when it gets called multiple times at the same time it will cause issues